app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -166,6 +167,29 @@ const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(supercla
         return batch;
     };
 
+    handleUpdateScheduledPostErrorCode = async ({scheduledPostId, errorCode, prepareRecordsOnly}: HandleScheduledPostErrorCodeArgs): Promise<ScheduledPostModel | null> => {
+        const database: Database = this.database;
+        const scheduledPost = await database.get<ScheduledPostModel>(SCHEDULED_POST).find(scheduledPostId);
+
+        if (!scheduledPost) {
+            logWarning(
+                `Scheduled Post with id ${scheduledPostId} not found in the database`,
+            );
+            return null;
+        }
+
+        const updatedScheduledPost = scheduledPost.prepareUpdate((record) => {
+            record.errorCode = errorCode;
+            record.updateAt = scheduledPost.updateAt; // We don't want to update the updateAt field
```

Comment:
If we don't want to update the updateAt, shouldn't we just not do it? I see value on the comment (but maybe it makes sense to add a bit more of reasoning), but no value on the assignation.